envelope.md

```diff
@@ -1,8 +1,8 @@
 # DSSE Envelope
 
-March 03, 2021
+May 10, 2024
 
-Version 1.0.0
+Version 1.1.0-draft
```

Comment:
But how are users to find the non-draft version? We don't have any convention for using (say) branches, and there's no documentation explaining this.

I think we need to either:
- (status quo) Say that anything living on `main` is "published", and commit to supporting it long-term.
- Come up with a way to publish draft versions, such as through git branches or by publishing on a website (most likely through GitHub Pages or Netlify).

Reply:
I think there's been drift in the status quo over time, because there was an explicit tag (https://github.com/secure-systems-lab/dsse/releases/tag/v1.0.0) for v1.0.0 with no patch releases for minor fixes since then. Do we commit to publishing on `main` henceforth and drop explicit tags? I do like the idea of having the tags, and perhaps we ought to clarify in the README to refer to them for releases?

Reply:
That would be fine too. It just needs to be documented and clearly communicated to readers. We would also want a branch for "latest" so that users don't always have to look up by tag. I think the cleanest solution is for `main` to always refer to the most recent version and `devel` (or similar) to point to the upcoming draft.

Reply:
That sounds reasonable to me. Here's what I propose:
* I'll open a separate PR against master that adds a note to the README about versioning
* I'll bump the versions of files on master now (before this PR) to v1.0.1 for the clarifications that have been made since v1.0.0 (I think a patch release is likely fine, skimming through the commits since then)

With this merged, we can tag v1.0.1 against master, which aligns with what you propose. I can then rebase this PR, drop the -draft from the docs, and change the base branch to devel.